TripScanner/src/main/java/com/tripscanner/TripScanner/controller/UserWebController.java

```diff
@@ -0,0 +1,109 @@
+package com.tripscanner.TripScanner.controller;
+
+import com.tripscanner.TripScanner.model.Destination;
+import com.tripscanner.TripScanner.model.Itinerary;
+import com.tripscanner.TripScanner.model.User;
+import com.tripscanner.TripScanner.service.DestinationService;
+import com.tripscanner.TripScanner.service.UserService;
+import com.tripscanner.TripScanner.service.ItineraryService;
+import org.hibernate.engine.jdbc.BlobProxy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Optional;
+
+@Controller
+public class UserWebController {
+
+    @Autowired
+    private ItineraryService itineraryService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private DestinationService destinationService;
+
+    @GetMapping("/user/{id}/image")
+    public ResponseEntity<Object> downloadImage(@PathVariable long id) throws SQLException {
+
+        Optional<User> user = userService.findById(id);
+        if (user.isPresent() && user.get().getImageFile() != null) {
+
+            Resource file = new InputStreamResource(user.get().getImageFile().getBinaryStream());
+
+            return ResponseEntity.ok().header(HttpHeaders.CONTENT_TYPE, "image/jpeg")
+                    .contentLength(user.get().getImageFile().length()).body(file);
+
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @GetMapping("/management/user/delete/{id}")
+    public String deleteUser(Model model, @PathVariable long id){
+        userService.delete(id);
+        return "redirect:/management/user/";
+    }
+
+    @GetMapping("/management/user/edit/{id}")
+    public String editUserIni(Model model, @PathVariable long id){
+        Optional<User> user = userService.findById(id);
+        model.addAttribute("mode", "edit");
+        model.addAttribute("edit", true);
+        model.addAttribute("type", "User");
+        model.addAttribute("add", false);
+        model.addAttribute("user", user.get());
```

Comment:
What is user? A whole instance of an User or the data of the current user?

Reply:
An instance of the user that the admin is editing, identified by his id